src/test/java/org/saltyrtc/client/tests/LazysodiumCryptoInstance.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2018 Threema GmbH
+ *
+ * Licensed under the Apache License, Version 2.0, <see LICENSE-APACHE file>
+ * or the MIT license <see LICENSE-MIT file>, at your option. This file may not be
+ * copied, modified, or distributed except according to those terms.
+ */
+package org.saltyrtc.client.tests;
+
+import com.goterl.lazycode.lazysodium.LazySodiumJava;
+import com.goterl.lazycode.lazysodium.SodiumJava;
+import com.goterl.lazycode.lazysodium.interfaces.Box;
+import org.saltyrtc.client.annotations.NonNull;
+import org.saltyrtc.client.crypto.CryptoException;
+import org.saltyrtc.client.crypto.CryptoInstance;
+import org.saltyrtc.client.crypto.CryptoProvider;
+
+import static com.goterl.lazycode.lazysodium.interfaces.Box.BEFORENMBYTES;
+import static com.goterl.lazycode.lazysodium.interfaces.Box.MACBYTES;
+
+/**
+ * An implementation of the CryptoInstance interface for lazysodium-java.
+ */
+public class LazysodiumCryptoInstance implements CryptoInstance {
+    @NonNull private final Box.Native sodium;
+    @NonNull private final byte[] k;
```

Comment:
Nit: I think we have a few spare bytes for a more descriptive name. :)

Reply:
I reused the terminology used by Lazysodium, but I guess you're right that it would be easier to read otherwise :slightly_smiling_face: